### PR TITLE
Tag based documentation generation

### PR DESCRIPTION
**Pull request by tw**
Added the ability to generate documentation based on a specific set of tags (provided as arguments). Updated a couple of packages. I also refactored the way processOperation was called to avoid using tag, but to iterate over the list of paths (IMO iterating over the tags is suboptimal). Hopefully no objections here...

### CODE REVIEW

File: src/main/java/org/rakam/client/builder/document/SlateDocumentGenerator.java [hunk 1]
```diff
@@ -152,19 +155,46 @@ private void buildSlateDocument()
...
 
-        if (!swagger.getTags().isEmpty()) {
-            for (Tag tag : swagger.getTags()) {
-                String name = tag.getName();
-                String description = tag.getDescription();
-                markdownBuilder.documentTitle(CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, name.replaceAll("-", " "))).newLine().textLine(description).newLine();
-                processOperation(name);
+        AtomicReference<String> previousTitle = new AtomicReference<>();
+        swagger.getPaths().forEach((path, model) -> {
+            String title = path.split("/")[1]; // TODO: this could be waaay better.
```
Comment (buremba) [inline src/main/java/org/rakam/client/builder/document/SlateDocumentGenerator.java:160]: There is also no guarantee that the paths are structured this way. Why do you think that tags are not the optimal way for determining the title?

**Comment by tw** [comment]
@buremba - any thoughts on this?

**Comment by buremba** [comment]
@tw , I already reviewed the PR and asked a question, could you please take a look at https://github.com/buremba/swagger-slate/pull/3/files/8a69c1528ac3a0a8df69cffac3011cf7d6fd0639#diff-ba570d9a834ec090565e8630b30f9b58 ?

**Comment by tw** [comment]
@buremba I cannot see the comment/question - can you please comment again?

**Comment by buremba** [comment]
@tw If you visit https://github.com/buremba/swagger-slate/pull/3/files/8a69c1528ac3a0a8df69cffac3011cf7d6fd0639#diff-ba570d9a834ec090565e8630b30f9b58 and search for "There is also no guarantee" in page you will see my message.

Also here the quote:
`There is also no guarantee that the paths are structured this way. Why do you think that tags are not the optimal way for determining the title?`

**Comment by buremba** [comment]
I just realized that I didn't send the PR review. Sorry about that.

**Pull request closed without merging**